### PR TITLE
Google Analytics - clarity on clientId

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Updated Google Analytics documentation to clarify how RudderStack takes `clientId`.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by akashrpo — APPROVED** [review]

File: destinations/analytics/google-analytics-ga.mdx [hunk 1]
```diff
@@ -455,6 +457,31 @@ ga(function (tracker) {
 
 On the server-side, if `clientId` is passed manually by sending `anonymousId`. If the **Send User ID** setting is enabled in the destination settings along with `cid` , `uid` will be sent where `userId` will be passed.
```
Comment (ishachopra0401) [inline destinations/analytics/google-analytics-ga.mdx:458]: I think something is missing in the first sentence.

File: destinations/analytics/google-analytics-ga.mdx [hunk 2]
```diff
@@ -455,6 +457,31 @@ ga(function (tracker) {
...
 
+### How RudderStack takes the `clientId`
+
+RudderStack takes the `clientId` in the following **decreasing** order of priority:
```
Comment (ishachopra0401) [inline destinations/analytics/google-analytics-ga.mdx:462]:
```suggestion
RudderStack takes the `clientId` in the following priority order:
```

File: destinations/analytics/google-analytics-ga.mdx [hunk 3]
```diff
@@ -455,6 +457,31 @@ ga(function (tracker) {
...
+
+1. If `clientId` is passed in the `integrations` object while initializing the SDK, as shown:
```
Comment (ishachopra0401) [inline destinations/analytics/google-analytics-ga.mdx:464]:
```suggestion
1. From the `integrations` object while initializing the SDK, as shown:
```

File: destinations/analytics/google-analytics-ga.mdx [hunk 4]
```diff
@@ -455,6 +457,31 @@ ga(function (tracker) {
...
+```javascript
+"integrations": {
+  "All": true,
+  "Google Analytics": {
+    "clientId": "clientId"
+  }
+},
+```
+
+2. Sets `gaExternalId` as the `clientId` if it is specified in the payload.
```
Comment (ishachopra0401) [inline destinations/analytics/google-analytics-ga.mdx:475]:
```suggestion
2. From the `gaExternalId` by setting it as the `clientId`, if specified in the payload.
```

File: destinations/analytics/google-analytics-ga.mdx [hunk 5]
```diff
@@ -455,6 +457,31 @@ ga(function (tracker) {
...
+3. If `anonymousId` is present.
```
Comment (ishachopra0401) [inline destinations/analytics/google-analytics-ga.mdx:476]:
```suggestion
3. From `anonymousId`, if present.
```

**Review by ishachopra0401 — CHANGES_REQUESTED** [review]

File: destinations/analytics/google-analytics-ga.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply destinations/analytics/google-analytics-ga.mdx:462]: with this change it's not explicitly clear if the priority is increasing or decreasing.

File: destinations/analytics/google-analytics-ga.mdx [hunk 6]
```diff
@@ -28,23 +28,21 @@ Before configuring your source and destination on the RudderStack app, please ch
...
-Please follow our guide on <a href="https://rudderstack.com/docs/connections/adding-source-and-destination-rudderstack/">How to Add a Source and Destination in RudderStack</a> to add a source and destination in RudderStack.
-
+Follow our guide on <a href="https://rudderstack.com/docs/connections/adding-source-and-destination-rudderstack/">Adding a Source and Destination in RudderStack</a> for more information.
 </div>
 
-Once the destination is enabled, events from our servers will start flowing to Google Analytics. You will start seeing the events in Google Analytics' real-time view.
+2. Assign a name to the destination and click on **Next**.
 
-Google Analytics' property `tracking id` is a required field to start sending events. Other inputs as shown below can be set as per your preferences:
+3. In the **Connection Settings**, **Tracking ID**  is a required field to set up the destination and start sending events. Other inputs as shown below can be set as per your preferences:
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply destinations/analytics/google-analytics-ga.mdx:45]: makes sense. will pick this up as a part of the doc revamp.

File: destinations/analytics/google-analytics-ga.mdx [hunk 7]
```diff
@@ -132,7 +134,7 @@ Google Analytics' universal tracking allows you to set a user ID to the identifi
 
 A sample `identify` call is as shown:
 
-```
+```javascript
 rudderanalytics.identify("abc123", {
```
Comment (ameypv-rudder) [inline destinations/analytics/google-analytics-ga.mdx:138]:
```suggestion
rudderanalytics.identify("user123", {
```

File: destinations/analytics/google-analytics-ga.mdx [hunk 8]
```diff
@@ -132,7 +134,7 @@ Google Analytics' universal tracking allows you to set a user ID to the identifi
...
         name: "Ruchira Moitra",
```
Comment (ameypv-rudder) [inline destinations/analytics/google-analytics-ga.mdx:139]:
```suggestion
        name: "Alex Keener",
```

File: destinations/analytics/google-analytics-ga.mdx [hunk 9]
```diff
@@ -132,7 +134,7 @@ Google Analytics' universal tracking allows you to set a user ID to the identifi
...
         email: "ruchira@gmail.com"
```
Comment (ameypv-rudder) [inline destinations/analytics/google-analytics-ga.mdx:140]:
```suggestion
        email: "alex@example.com"
```

File: destinations/analytics/google-analytics-ga.mdx (hunk 1 above)
Comment (ameypv-rudder) [inline destinations/analytics/google-analytics-ga.mdx:458]:
```suggestion
For server-side calls, if `clientId` is passed manually by sending `anonymousId` and if the **Send User ID** setting is enabled in the destination settings along with `cid`, then RudderStack passes the `userId` as the `uid`.
```

File: destinations/analytics/google-analytics-ga.mdx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply destinations/analytics/google-analytics-ga.mdx]: done!

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**